addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/handler/RouteHandler.java

```diff
@@ -0,0 +1,278 @@
+        props.put(ROUTE_PROPERTY_SHORT_NAME, data.routeShortName);
+        updateProperties(props);
+    }
+
+    /**
+     * Publishes the channel with data and possibly schedules work to update it again when the next event has passed.
+     */
+    private void publishChannel(ChannelUID channelUID, Calendar now, List<ArrivalAndDeparture> arrivalAndDepartures) {
+        ChannelConfig channelConfig = getThing().getChannel(channelUID.getId()).getConfiguration()
+                .as(ChannelConfig.class);
+        long offsetMs = TimeUnit.SECONDS.toMillis(channelConfig.getOffset());
+        for (int i = 0; i < arrivalAndDepartures.size(); i++) {
+            ArrivalAndDeparture data = arrivalAndDepartures.get(i);
+            long time;
+            switch (channelUID.getId()) {
+                case CHANNEL_ID_ARRIVAL:
+                    time = data.predicted ? data.predictedArrivalTime : data.scheduledArrivalTime;
+                    break;
+                case CHANNEL_ID_DEPARTURE:
+                    time = data.predicted ? data.predictedDepartureTime : data.scheduledDepartureTime;
+                    break;
+                case CHANNEL_ID_UPDATE:
+                    time = data.lastUpdateTime;
+                    break;
+                default:
+                    logger.warn("No code to handle publishing to {}", channelUID.getId());
+                    return;
+            }
+
+            // Do not publish this if it's already passed.
+            Calendar cal = (new Calendar.Builder()).setInstant(time - offsetMs).build();
+            if (cal.before(now)) {
+                logger.debug("Not notifying {} because it is in the past.", channelUID.getId());
+                continue;
+            }
+            updateState(channelUID, new DateTimeType(cal));
+
+            // Update properties only when we update arrival information. This is not perfect.
+            if (channelUID.getId().equals(CHANNEL_ID_ARRIVAL)) {
+                updatePropertiesFromArrivalAndDeparture(data);
+            }
+
+            // Schedule updates in the future. These may be canceled if we are notified about new data in the future.
+            List<ArrivalAndDeparture> remaining = arrivalAndDepartures.subList(i + 1, arrivalAndDepartures.size());
+            if (remaining.isEmpty()) {
+                return;
+            }
+            scheduledFutures.add(scheduler.schedule(new Runnable() {
+                @Override
+                public void run() {
+                    publishChannel(channelUID, Calendar.getInstance(), remaining);
+                }
+            }, time - now.getTimeInMillis(), TimeUnit.MILLISECONDS));
+            return;
+        }
+    }
+
+    private void scheduleTriggerEvents(ChannelUID channelUID, Calendar now,
+            List<ArrivalAndDeparture> arrivalAndDepartures) {
+        scheduleTriggerEvents(channelUID, now, arrivalAndDepartures, new Function<ArrivalAndDeparture, Long>() {
```

Comment:
No problem, its a nice addition you definitely consider to read about them later, it's a not blocker here.